selenium-pom/src/main/java/uk/sponte/automation/seleniumpom/proxies/handlers/PageElementListHandler.java

```diff
@@ -137,11 +143,11 @@ public void refresh() {
 
         for (int i = 0; i < elements.size(); i++) {
             WebElement e = elements.get(i);
-            Object s = webElements.get(i);
 
-            if (s == null) {
-                webElements.set(i, getPageElementProxy(e));
+            if (webElements.size() == i) {
```

Comment:
This does not need greater than as it works correctly and also removing is done before the loop now